src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/migration/MigrateVideoBookingServiceTest.kt

```diff
@@ -0,0 +1,798 @@
+            comment = "Probation booking delete event comments",
+            courtCode = BLACKPOOL_MC_PPOC,
+            preStartTime = null,
+            preEndTime = null,
+            mainStartTime = yesterday().atTime(10, 0),
+            mainEndTime = yesterday().atTime(11, 0),
+            postStartTime = null,
+            postEndTime = null,
+            prisonCode = WERRINGTON,
+            createdByUsername = "MIGRATION PROBATION DELETE USER",
+            courtName = null,
+            madeByTheCourt = true,
+            preLocationId = null,
+            mainLocationId = 1,
+            postLocationId = null,
+          ),
+        ),
+      ),
+    )
+
+    inOrder(videoBookingRepository, bookingHistoryRepository) {
+      verify(videoBookingRepository).saveAndFlush(videoBookingCaptor.capture())
+      verify(bookingHistoryRepository, times(2)).saveAndFlush(bookingHistoryCaptor.capture())
+    }
+
+    bookingHistoryCaptor.allValues hasSize 2
+
+    with(bookingHistoryCaptor.firstValue) {
+      historyType isEqualTo HistoryType.CREATE
+      createdBy isEqualTo "MIGRATION PROBATION CREATE USER"
+      createdTime isEqualTo 2.daysAgo().atStartOfDay()
+
+      appointments().single()
+        .isAtPrison(WERRINGTON)
+        .isForPrisonerNumber("DEF123")
+        .isOnDate(yesterday())
+        .startsAt(LocalTime.of(10, 0))
+        .endsAt(LocalTime.of(11, 0))
+        .isForAppointmentType("VLB_PROBATION")
+        .isAtLocation(werringtonLocation)
+    }
+
+    with(bookingHistoryCaptor.secondValue) {
+      historyType isEqualTo HistoryType.CANCEL
+      createdBy isEqualTo "MIGRATION PROBATION DELETE USER"
+      createdTime isEqualTo 1.daysAgo().atStartOfDay()
+
+      appointments().single()
+        .isAtPrison(WERRINGTON)
+        .isForPrisonerNumber("DEF123")
+        .isOnDate(yesterday())
+        .startsAt(LocalTime.of(10, 0))
+        .endsAt(LocalTime.of(11, 0))
+        .isForAppointmentType("VLB_PROBATION")
+        .isAtLocation(werringtonLocation)
+    }
+  }
+
+  private fun BookingHistoryAppointment.isAtPrison(prisonCode: String) = also { it.prisonCode isEqualTo prisonCode }
+
+  private fun BookingHistoryAppointment.isForPrisonerNumber(prisonerNumber: String) =
+    also { it.prisonerNumber isEqualTo prisonerNumber }
+
+  private fun BookingHistoryAppointment.isOnDate(appointmentDate: LocalDate) =
+    also { it.appointmentDate isEqualTo appointmentDate }
+
+  private fun BookingHistoryAppointment.startsAt(startTime: LocalTime) = also { it.startTime isEqualTo startTime }
+
+  private fun BookingHistoryAppointment.endsAt(endTime: LocalTime) = also { it.endTime isEqualTo endTime }
+
+  private fun BookingHistoryAppointment.isForAppointmentType(appointmentType: String) =
+    also { it.appointmentType isEqualTo appointmentType }
+
+  private fun BookingHistoryAppointment.isAtLocation(location: Location) =
+    also { it.prisonLocKey isEqualTo location.key }
+
```

Comment:
As this is a run-once feature, we could maybe limit the numbers of automated tests a little? And instead rely on our test runs in preprod a little more?

Reply:
Perhaps.  Just aware it is easier to test here (at the moment anyway) than getting PR's out, merged, DB's cleared etc.